### PR TITLE
JS: Convert specs to use project fixtures

### PR DESCRIPTION
**Pull request by feelepxyz**
Converting npm/yarn specs to use project based fixtures:

- `YarnLockfileParser`
- `PathDependencyBuilder`
- `FileUpdater`
- ` LibraryDetector`
- `DependencyFilesBuilder`
- `ConflictingDependencyResolver`

### CODE REVIEW

File: npm_and_yarn/spec/dependabot/npm_and_yarn/update_checker/dependency_files_builder_spec.rb [hunk 1]
```diff
@@ -50,47 +35,61 @@
...
-    )
-  end
-
   describe "#write_temporary_dependency_files" do
     it "writes the relevant files to disk" do
       Dependabot::SharedHelpers.in_a_temporary_directory do
         builder.write_temporary_dependency_files
 
         expect(Dir.glob("*")).to match_array(
-          %w(package.json package-lock.json yarn.lock npm-shrinkwrap.json)
```
Comment (feelepxyz) [inline npm_and_yarn/spec/dependabot/npm_and_yarn/update_checker/dependency_files_builder_spec.rb:66]: Moved the shrinkwrap test to use a separate project

File: npm_and_yarn/spec/dependabot/npm_and_yarn/dependency_files_filterer_spec.rb [hunk 2]
```diff
@@ -14,117 +14,95 @@
   end
```
Comment (feelepxyz) [inline npm_and_yarn/spec/dependabot/npm_and_yarn/dependency_files_filterer_spec.rb:14]: Ended up re-structuring this spec a fair bit as I found it hard to follow, de-nested some contexts and used existing projects and added a npm 7 workspace spec.

File: npm_and_yarn/spec/dependabot/npm_and_yarn/dependency_files_filterer_spec.rb [hunk 3]
```diff
@@ -14,117 +14,95 @@
...
+    dependency_files.find { |f| f.name == file_name }
   end
 
   describe ".files_requiring_update" do
     it do
-      is_expected.to contain_exactly(package_json, yarn_lock, npm_lock)
+      is_expected.to contain_exactly(
+        project_dependency_file("package.json"),
+        project_dependency_file("package-lock.json"),
+        project_dependency_file("yarn.lock")
```
Comment (jurre) [inline npm_and_yarn/spec/dependabot/npm_and_yarn/dependency_files_filterer_spec.rb]: This is so much clearer, thanks

**Review by jurre — APPROVED** [review]
Thanks for tackling these

File: npm_and_yarn/spec/dependabot/npm_and_yarn/dependency_files_filterer_spec.rb (hunk 3 above)
Comment (feelepxyz) [reply npm_and_yarn/spec/dependabot/npm_and_yarn/dependency_files_filterer_spec.rb]: Ops just moved this to a different PR to make it easier to review: https://github.com/dependabot/dependabot-core/pull/3148
